### PR TITLE
fix(FlipView): not moving views

### LINKED ISSUE #7671: [Skia] [Wasm] FlipView tests fail for managed implementation
### Current behavior

FlipView tests failing, and control doesn't currently seem to work on Wasm or Skia.

### Expected behavior

Control works.

### How to reproduce it (as minimally and precisely as possible)

At least two failures were noted:

## Runtime test:

1. Re-enable `Given_FlipView.When_Observable_ItemsSource_And_Added()`
2. Note that it fails on Skia and Wasm.

Initial investigation: on adding an item to the items source collection, the scroll is updated, but it appears that items are not fully laid out yet, and the snap points logic applies an incorrect scroll, which in turn triggers an incorrect SelectedIndex to be set.

## Sample

1. Run SamplesApp.
2. Navigate to FlipView_Button sample
3. Try to flip the FlipView
4. Notice that Button 2 doesn't appear

Note: the associated UI tests (`FlipView_Tests`) are passing, but apparently spuriously - the test simply calls WaitForElement() on Button 2, but since the items panel is a StackPanel (not virtualized), Button 2 is already present (but not visible). The test should be amended to take a screenshot and check a pixel value to make sure that the flip really happened.

### Workaround

_No response_

### Works on UWP/WinUI

_No response_

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

WebAssembly, Skia (WPF), Skia (GTK on Linux/macOS/Windows), Skia (Tizen)

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### LINKED ISSUE #8353: `FlipView` does not flip on Skia
### Current behavior

flip

### Expected behavior

flip2

### How to reproduce it (as minimally and precisely as possible)

`FlipView` samples in Samples app

### Workaround

-

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia, Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

Latest dev

### Affected platforms

Skia (WPF), Skia (GTK on Linux/macOS/Windows), Skia (Tizen)

### IDE

Visual Studio 2022

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): closes #8353 

fixes #7671 

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?
FlipView was not changing view (forward or backward) on either WASM or Skia

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.FluentTheme.v1/Resources/Version1/PriorityDefault/FlipView_themeresources.xaml [hunk 1]
```diff
@@ -320,6 +320,7 @@
                             ZoomMode="Disabled"
                             TabNavigation="{TemplateBinding TabNavigation}"
                             IsTabStop="False"
+                            ForceChangeToCurrentView="True"
```
Comment (jeromelaban) [inline src/Uno.UI.FluentTheme.v1/Resources/Version1/PriorityDefault/FlipView_themeresources.xaml:323]: Is this coming from the official style?

File: src/Uno.UI.FluentTheme.v2/Resources/Version2/PriorityDefault/FlipView_themeresources.xaml [hunk 2]
```diff
@@ -394,6 +394,7 @@
                             ZoomMode="Disabled"
                             TabNavigation="{TemplateBinding TabNavigation}"
                             IsTabStop="False"
+							ForceChangeToCurrentView="True"
```
Comment (jeromelaban) [inline src/Uno.UI.FluentTheme.v2/Resources/Version2/PriorityDefault/FlipView_themeresources.xaml:397]: Same here.

File: src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.cs [hunk 3]
```diff
@@ -61,12 +61,19 @@ public object ScrollOwner
...
 #if __WASM__
 		bool _forceChangeToCurrentView;
-
 		bool IScrollContentPresenter.ForceChangeToCurrentView
 		{
 			get => _forceChangeToCurrentView;
 			set => _forceChangeToCurrentView = value;
 		}
+
+#elif __SKIA__
```
Comment (MartinZikmund) [inline src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.cs:70]: Why can't we use `#if __WASM__ || __SKIA__` above?

File: src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.Managed.cs [hunk 4]
```diff
@@ -75,10 +75,28 @@ private static double GetHorizontalScrollWheelDelta(Size size, double delta)
...
-		
-		public bool CanHorizontallyScroll { get; set; }
 
-		public bool CanVerticallyScroll { get; set; }
+		private bool _canHorizontallyScroll;
+		public bool CanHorizontallyScroll
+		{
+			get => _canHorizontallyScroll
+#if __SKIA__
+			|| _forceChangeToCurrentView
```
[the remark this answers is not in the document]
Comment (ajpinedam) [reply src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.Managed.cs:84]: In the case of WASM is coming from the ScrollContentPresenter.wasm.cs

File: src/Uno.UI.FluentTheme.v1/Resources/Version1/PriorityDefault/FlipView_themeresources.xaml (hunk 1 above)
Comment (ajpinedam) [reply src/Uno.UI.FluentTheme.v1/Resources/Version1/PriorityDefault/FlipView_themeresources.xaml]: removed

File: src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.cs (hunk 3 above)
Comment (ajpinedam) [reply src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/ScrollContentPresenter.cs:71]: Only to avoid making this property public. 

For WASM, there's an interface `IScrollContentPresenter.cs`, and we do an Explicit implementation to conform to the interface signature.

**Review by MartinZikmund — APPROVED** [review]

File: src/Uno.UI.FluentTheme.v2/Resources/Version2/PriorityDefault/FlipView_themeresources.xaml [hunk 5]
```diff
@@ -14,15 +14,15 @@
         <ResourceDictionary x:Key="Default">
```
Comment (Xiaoy312) [inline src/Uno.UI.FluentTheme.v2/Resources/Version2/PriorityDefault/FlipView_themeresources.xaml:14]: unnecessary changes

**Review by Xiaoy312 — APPROVED** [review]

**Review by jeromelaban — APPROVED** [review]
Thanks @ajpinedam!

**Comment by jeromelaban** [comment]
The issue is coming from nuget's CDN, we'll retry the builds.
